src/hooks/useOutsideClick/types.ts

```diff
@@ -0,0 +1,8 @@
+import type { RefObject } from 'react';
+
+export type AnyEvent = MouseEvent | TouchEvent;
+
+export type OutsideClickProps = <T extends HTMLElement = HTMLElement>(
+  ref: RefObject<T>,
+  handler: (event: AnyEvent) => void,
+) => void;
```

Comment:
Types file not need, per me and Yuri's last PR, if only a few lines of types, don't want to work in 2 files and does not require a directory.

ex:

```
import { useOutsideClick } from 'hooks/useOutsideClick';
import type { OutsideClickProps } from 'hooks/useOutsideClick/types';
```

vs. just 1 liner:

```
import { useOutsideClick, OutsideClickProps } from 'hooks/useOutsideClick';
```

I think 1 liner is simpler to import and use.